### PR TITLE
#15 finish route api/leaderboard/queue

### PR DESCRIPTION
**Pull request by bnonni**
Method: `GET`
Query:
- `hostShortName="atl"`: required
- `next=[true | false]`: optional
Returns: 
- if `next=false` or not present, sorts "queued" plays for hostId by `runningTotal` and `queueTimestamp` as tie breaker
- if `next=true`, updates the status of the leading play from `queued` to `next` in the database & returns the queue of plays sorted by `runningTotal` and `queueTimestamp` as tie breaker

TODO:
- [x] Finalize api route data model - using shortName vs. customerId, which are/aren't required fields, instance id or not, how uuids are created
- [x] Implement final logic
- [x] Test route
- [ ] Merge to main

### CODE REVIEW

File: pages/api/bid/submit.ts [hunk 1]
```diff
@@ -1,53 +1,74 @@
-import { NextApiRequest, NextApiResponse } from "next";
-import cuid from "cuid";
-import Hosts from "../../../models/Host";
-import Users from "../../../models/User";
-import Plays, { Play } from "../../../models/Play";
-import { Bid } from "../../../models/Bid";
-import connectDB from "../../../middleware/mongodb";
+import { NextApiRequest, NextApiResponse } from 'next';
+import cuid from 'cuid';
+import Customers from '../../../models/Host';
```
Comment (alexlwn123) [inline pages/api/bid/submit.ts:3]: This should be "Host", right?

**Review by alexlwn123 — COMMENTED** [review]
Looks great, Thanks!

**Review by alexlwn123 — APPROVED** [review]
Looks great, Thanks! 

Closes #15

File: pages/api/bid/submit.ts (hunk 1 above)
Comment (bnonni) [reply pages/api/bid/submit.ts:3]: Right. Will submit in a cleanup PR.

File: models/Play.ts [hunk 2]
```diff
@@ -7,58 +7,58 @@ import { Bid } from './Bid';
...
+    },
+    queueTimestamp: {
+        type: String,
+        required: true,
+    },
+    playedTimestamp: {
+        type: String,
+    },
+    bids: {
+        type: new Array<Bid>(),
```
[the remark this answers is not in the document]
Comment (bnonni) [reply models/Play.ts:55]: Makes sense. Will edit and submit in cleanup PR.

[the remark this answers is not in the document]
Comment (bnonni) [reply models/Play.ts:55]: Honestly, not entirely sure how to do this, so I'll let you do it.

**Pull request merged**